### PR TITLE
AV-183081 Ingestion layer UT for Gateway

### PR DESCRIPTION
**Pull request by aaha97**
This PR includes

1. added unit tests for gateway and gateway class (Create, Update and Delete)
2. object store for gateway and gateway class
3. gateway class validation. check if controller is set to AKO
4. code cleanup

### CODE REVIEW

**Comment by aaha97** [comment]
`=== RUN   TestGatewayCUD
2023-07-17T11:27:37.163+0530    INFO    utils/queue.go:121      Worker id 1
    gateway_test.go:127: Created gateway gw-example
    gateway_test.go:136: Updated gateway gw-example
    gateway_test.go:144: Deleted gateway gw-example
--- PASS: TestGatewayCUD (0.02s)
=== RUN   TestGatewaClassyCUD
    gateway_test.go:168: Created gateway class gw-class-example
    gateway_test.go:177: Updated gateway class gw-class-example
    gateway_test.go:185: Deleted gateway class gw-class-example
--- PASS: TestGatewaClassyCUD (0.02s)
PASS`

File: ako-gateway-api/tests/gateway_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	gatewayClass := gatewayv1beta1.GatewayClass{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "gateway.networking.k8s.io/v1beta1",
+			Kind:       "GatewayClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "gw-class-example",
+		},
+		Spec: gatewayv1beta1.GatewayClassSpec{
+			ControllerName: "ako.vmware.com/avi-lb",
```
Comment (swathinsankaran) [inline ako-gateway-api/tests/gateway_test.go:158]: Are we having a validation logic to ignore the gatewayclass if the controllerName is not given as `"ako.vmware.com/avi-lb"`? Shall we add a UT for that in the next PR?

**Review by swathinsankaran — APPROVED** [review]
LGTM

**Comment by aaha97** [comment]
`=== RUN   TestGatewayCUD
    gateway_test.go:127: Created gw-example
    gateway_test.go:136: Updated gw-example
    gateway_test.go:144: Deleted gw-example
--- PASS: TestGatewayCUD (0.02s)
=== RUN   TestGatewaClassyCUD
    gateway_test.go:168: Created gw-class-example
    gateway_test.go:178: Updated gw-class-example
    gateway_test.go:186: Deleted gw-class-example
--- PASS: TestGatewaClassyCUD (0.02s)
PASS`

File: ako-gateway-api/objects/gateway_store.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		//TODO update GatewayToGatewayClass and GatewayClassToGateway
+	}
+}
+
+func (g *GWLister) GetGatewayClassToGateway(gwClass string) []string {
+	g.gwLock.Lock()
+	defer g.gwLock.Unlock()
+
+	_, gatewayList := g.GatewayClassToGateway.Get(gwClass)
+	return gatewayList.([]string)
```
Comment (swathinsankaran) [inline ako-gateway-api/objects/gateway_store.go:82]: The gatewayList item should contain `namespace/gatewayname` for easier mapping.

File: ako-gateway-api/objects/gateway_store.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+type GWLister struct {
+	gwLock sync.RWMutex
+
+	//Gateways with AKO as controller
+	GatewayClassStore *objects.ObjectMapStore
+
+	//Gateway -> GatewayClass
+	GatewayToGatewayClass *objects.ObjectMapStore
+
+	//GatewayClass -> [gateway1, gateway2]
```
Comment (swathinsankaran) [inline ako-gateway-api/objects/gateway_store.go:46]: namespace also must be included since the gateway is a namespace scoped object.

**Pull request merged**